src/v/cluster/members_manager.cc

```diff
@@ -423,6 +451,49 @@ void members_manager::join_raft0() {
     });
 }
 
+bool members_manager::try_register_node_id(
+  const model::node_id& requested_node_id,
+  const model::node_uuid& requested_node_uuid) {
+    vassert(requested_node_id != model::node_id(-1), "invalid node ID");
+    const auto it = _id_by_uuid.find(requested_node_uuid);
+    if (it == _id_by_uuid.end()) {
+        if (_members_table.local().contains(requested_node_id)) {
```

Comment:
just to make sure if i understand this correctly, if a node running new redpanda version that was already part of the cluster and have node_id set in configuration file will have a data directory wiped it will regenerate UUID and will be remapped to new UUID ?

Reply:
If you don't mind, I will do this in a follow-up change, so we can merge this sooner.